libtiledbsoma/src/common/soma_column_selection.h

```diff
@@ -0,0 +1,99 @@
+/**
+ * @file   soma_coordinate_selection.h
+ *
+ * @section LICENSE
+ *
+ * Licensed under the MIT License.
+ * Copyright (c) TileDB, Inc. and The Chan Zuckerberg Initiative Foundation
+ *
+ * @section DESCRIPTION
+ *
+ * This defines the coordinate seletion API.
+ *
+ */
+
+#ifndef SOMA_COORDINATE_SELECTION_H
+#define SOMA_COORDINATE_SELECTION_H
+
+#include <math.h>
+#include <span>
+#include <sstream>
+#include <string>
+#include <utility>
+#include <variant>
+
+#include "../utils/common.h"
+
+namespace tiledbsoma {
+
+template <typename T>
+struct SOMASliceSelection {
+   public:
+    /**
+     * Creates a closed slice (includes both end points).
+     *
+     * @param slice_start The first value (inclusive) of the slice.
+     * @param slice_stop The last value (inclusive) of the slice.
+     */
+    SOMASliceSelection(T slice_start, T slice_stop)
+        : start{slice_start}
+        , stop{slice_stop} {
+        // Using sstream because we don't want to include fmt directly in external header.
+        std::stringstream ss;
```

Comment:
suggest moving `ss` create & init inside the if statement scope, so it is only done when there is an actual failure.